src/Common/EventStore/FollowEventStoreDispatcher.php

```diff
@@ -45,11 +38,19 @@ public function dispatch(int $batchSize): void
             $batchSize
         );
 
-        if (!empty($storedEvents)) {
-            $this->storedEventPublisher->publish($storedEvents);
+        if (count($storedEvents) === 0) {
+            return 0;
+        }
 
-            $lastStoredEventId = end($storedEvents)->id();
-            $this->eventStorePointer->trackMostRecentPublishedStoredEventId($lastStoredEventId);
+        foreach ($storedEvents as $storedEvent) {
+            $this->storedEventSubscriber->handle($storedEvent);
         }
+
+        ($this->afterHandlingStoredEvents)();
```

Comment:
An explicit handling of this in the `StoredEventSubscriber` is missing. E.g., when using publisher confirms with RabbitMQ or flushing techniques with other messages queues to ensure that packets arrived at the server, this should explicitly be handled by the `StoredEventSubscriber`. However, this works for now and will be changed with another feature as this pull request is big already and this is change seems unrelated.